MapboxNavigation/LanesView.swift

```diff
@@ -66,31 +66,33 @@ open class LanesView: UIView {
         separatorView.trailingAnchor.constraint(equalTo: trailingAnchor).isActive = true
     }
     
-    func update(for currentLegProgress: RouteLegProgress) {
-        guard let step = currentLegProgress.upComingStep else { return }
-        guard !currentLegProgress.userHasArrivedAtWaypoint else { return }
-        let durationRemaining = currentLegProgress.currentStepProgress.durationRemaining
-        
+    func update(for currentStepProgress: RouteStepProgress) {
         clearLaneViews()
         
-        if let allLanes = step.intersections?.first?.approachLanes,
-            let usableLanes = step.intersections?.first?.usableApproachLanes,
-            durationRemaining < RouteControllerMediumAlertInterval {
-            
-            for (i, lane) in allLanes.enumerated() {
-                let laneView = laneArrowView()
-                laneView.lane = lane
-                laneView.maneuverDirection = step.maneuverDirection
-                laneView.isValid = usableLanes.contains(i as Int)
-                stackView.addArrangedSubview(laneView)
-            }
+        let distanceTraveled = currentStepProgress.distanceTraveled
+        guard let visualInstruction = currentStepProgress.currentVisualInstruction,
+              let tertiaryInstruction = visualInstruction.tertiaryInstruction,
+                  distanceTraveled <= visualInstruction.distanceAlongStep,
```

Comment:
This should not be necessary, this check should already be happening via in RouteController.swift and we should be using the currentVisualInstructionIndex, similar to how the main instruction is getting updated: 

https://github.com/mapbox/mapbox-navigation-ios/blob/f08eb8ee299dd8b8e71bf0c72adaa64c6c4dd174/MapboxNavigation/RouteMapViewController.swift#L345